term/src/input.rs

```diff
@@ -36,8 +36,8 @@ pub struct MouseEvent {
     pub kind: MouseEventKind,
     pub x: usize,
     pub y: VisibleRowIndex,
-    pub x_pixel_offset: usize,
-    pub y_pixel_offset: usize,
+    pub x_pixel_offset: isize,
```

Comment:
I don't have time for a full review right now, but this caught my eye: why is the type changing? I'm pretty sure that this can never legitimately be negative. @AutumnMeowMeow: can you confirm?

Reply:
The idea is for them to be able to have negative values, to encode information about the mouse leaving the area. For instance, y = 0 meaning that the cell is the top one, but y pixel offset = -10 means the mouse is 10 pixels above that cell.

Reply:
(Sorry if incoherent, watching Matrix 4 while drunk... 🍷🔫🔫🔫👭🔫🔫🔫🔫🔫🔫🔫🔫🔫👭🔫🔫🔫🔫🔫🔫🔫🔫🔫💞💞)

The terminal should not be reporting negative values (they are required to be positive in CSI sequences), but at one point xterm itself was reporting negative when the mouse was captured (button down) and dragged above the top-left corner of the text area -- when using SGR-Pixel mode. I didn't report that as a bug, someone else might have.  Let's see if I can find that thread....  here: https://github.com/contour-terminal/contour/issues/574#issuecomment-1008298182  Of course j4james is the real expert. :-)

Within wezterm, the offset values could work the same whether positive or negative values.

Reply:
Note that the negative offset is only used internally/where it makes sense, in other places I've clipped it to positive only values, like here: https://github.com/wez/wezterm/blob/be567fb8116419537b5957c761c663662ea62c3b/term/src/terminalstate/mouse.rs#L72

Reply:
On the topic of those offsets, they have an unexpected (to me) behavior, so I'm not sure if it's a bug or could have weird side effects. These are the values reported:

![wezterm_cell_reporting](https://user-images.githubusercontent.com/1418372/161377370-17ccb054-3c1b-48c3-87cf-1ff2df28ef51.svg)

The blue cells are what is reported in the `x` property, because of the rounding here: https://github.com/wez/wezterm/blob/be567fb8116419537b5957c761c663662ea62c3b/wezterm-gui/src/termwindow/mouseevent.rs#L85-L87, which is done so mouse selection feel more natural, yet the `x_pixel_offset` is calculated based on the real cell, so the reported `cell, offset` are not in sync.

I'm thinking maybe it would be better to report the real cell and offset, and let the code that's using them do the relevant processing.

Reply:
@AutumnMeowMeow: what do you think about these positioning offsets; are they actually what you want to see in the report sent to your app?